src/_pytest/_code/source.py

```diff
@@ -237,9 +237,7 @@ def getfslineno(obj):
 def findsource(obj):
     try:
         sourcelines, lineno = inspect.findsource(obj)
-    except py.builtin._sysex:
-        raise
-    except:  # noqa
+    except Exception:
```

Comment:
pytest should already have a correct alias for that we ought not to remove the raises case

Reply:
These end up the same, catching and reraising non-Exception is the same as just letting it raise

Reply:
(the edge cases are those which extend BaseException and not Exception such as KeyboardInterrupt and SystemExit)

Reply:
@asottile yikes, thats easy to miss at first, thanks for providing context